packages/wrangler/src/proxy.ts

```diff
@@ -207,18 +213,18 @@ export function usePreviewServer({
       const { headers, url } = message;
       addCfPreviewTokenHeader(headers, previewToken.value);
       headers["host"] = previewToken.host;
-      const localWebsocket = new WebSocket(message, socket, body);
+      const localWebsocket = new WebSocket(message, socket, body) as IWebsocket;
       // TODO(soon): Custom WebSocket protocol is not working?
       const remoteWebsocketClient = new WebSocket.Client(
         `wss://${previewToken.host}${url}`,
         [],
         { headers }
-      );
+      ) as IWebsocket;
```

Comment:
does both the client and regular socket have the same type?

Reply:
Yes, as far as I could discern from the Faye docs the types were the same the only difference being abstraction of `.on("message")`

Reply:
Just doubled checked they also have this mention in their README 

> It does not provide any abstraction other than the standard [WebSocket API](https://html.spec.whatwg.org/multipage/comms.html#network).